openslides_backend/migrations/migrations/0029_remove_permission_user_can_see_extra_data.py

```diff
@@ -0,0 +1,44 @@
+from typing import List, Optional
+
+from datastore.migrations.core.base_migration import BaseMigration
+from datastore.migrations.core.events import (
+    BaseEvent,
+    CreateEvent,
+    ListUpdateEvent,
+    UpdateEvent,
+)
+from datastore.shared.util import collection_from_fqid
+
+
+class Migration(BaseMigration):
+    """
+    This migration removes the 'user.can_see_extra_data' permission from
+    groups.
+    """
+
+    target_migration_index = 30
+    group_permission = "user.can_see_extra_data"
+
+    def migrate_event(
+        self,
+        event: BaseEvent,
+    ) -> Optional[List[BaseEvent]]:
+        collection = collection_from_fqid(event.fqid)
+        if collection != "group":
+            return None
+
+        if isinstance(event, (CreateEvent, UpdateEvent)):
+            if self.group_permission in event.data.get("permissions", []):
+                event.data["permissions"].remove(self.group_permission)
+                return [event]
+        elif isinstance(event, ListUpdateEvent):
+            mark = False
```

Comment:
you can remove this `mark` handling and just always return `[event]`, since this is just what's used internally in case `None` is returned.